doc/develop/test/twister.rst

```diff
@@ -780,6 +780,35 @@ filter: <expression>
 
     Would match it.
 
+required_images: <list of images>
+    A list of test scenarios that are required to be built for this test to run.
+    If the test does not have a ``CMakeLists.txt`` file or has set ``no_own_image``
+    to True, then the first image from this list will be used as the test
+    scenario image.
+    All required test scenarios must be available from the tree (given with ``-T`` option).
+    If required test scenario is not available in tree or build is failed,
+    the test will be skipped.
+    If the required scenario is prefixed with a platform name, then it will be
+    build for that platform only.
+    The following is an example yaml with 2 required images.
+
+    .. code-block:: yaml
+
+        tests:
+          shared.app.example:
+            required_images:
```

Comment:
both image will build and how to flash them?

Reply:
Flashing is done by Twister as it is now. If the test scenario does not have own image, then build_dir of first required_image is copied to the build_dir of the current test, and is used to flash the board. The other required images are only to use in custom test implementation. For pytest-harness I pass paths to build_dirs of required images. Then one can use those paths to get artefacts, to use them in custom pytest testcases (binaries for upgrades, or to flash another board that is connected to our DUT ... but flashing of second board must be added in custom script)